src/rules/no_func_assign.rs

```diff
@@ -30,6 +30,44 @@ impl LintRule for NoFuncAssign {
     let mut visitor = NoFuncAssignVisitor::new(context);
     visitor.visit_program(program, program);
   }
+
+  fn docs(&self) -> &'static str {
+    r#"Disallows the overwriting/reassignment of an existing function
+
+Javascript allows for the reassignment of a function definition.  This is
+generally a mistake on the developers part, or poor coding practice as code
+readability and maintainability will suffer.
+    
+### Invalid:
+```typescript
+function foo() {}
+foo = bar;
+
+const a = function baz() {
+  baz = "now I'm a string";
+}
+
+myFunc = existingFunc;
+function myFunc() {}
+```
+
+### Valid:
+```typescript
+function foo() {}
+const someVar = bar;
```

Comment:
```suggestion
function foo() {}
const someVar = foo;
```

It would be nicer to show that we are allowed to assign a function to other variables.